src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -3289,24 +3331,58 @@ private static bool TryGetNumberValue(string strNum, bool hex, bool real, char s
                         {
                             multiplier = -multiplier;
                         }
+
                         strNum = strNum.Substring(1);
                     }
+
                     style = hex ? NumberStyles.AllowHexSpecifier : NumberStyles.AllowLeadingSign;
 
                     long longValue;
-                    if (suffix == 'l' || suffix == 'L')
+                    switch (suffix)
                     {
-                        if (long.TryParse(strNum, style, NumberFormatInfo.InvariantInfo, out longValue))
-                        {
-                            result = longValue * multiplier;
-                            return true;
-                        }
-                        result = null;
-                        return false;
+                        case NumberSuffixFlags.Long:
+                            if (long.TryParse(strNum, style, NumberFormatInfo.InvariantInfo, out longValue))
+                            {
+                                result = longValue * multiplier;
+                                return true;
+                            }
+
+                            result = null;
+                            return false;
+                        case NumberSuffixFlags.Short:
+                            if (short.TryParse(strNum, style, NumberFormatInfo.InvariantInfo, out short s))
```

Comment:
> , out short s)

nit: it would be great if you can change rest `out` param uses in this method to use pattern matching as well. But you don't have to, no blocking at all.

Reply:
I'll make sure I address this in the follow up. I have a lot of those replaced, but I'll do an extra pass just to be sure!

Reply:
@vexx32 You doesn't still fix all cases in _the method_. (See lines 3255, 3578, 3346 ..........)